devicetypes/smartthings/fidure-2nd-generation-thermostat.src/fidure-2nd-generation-thermostat.groovy

```diff
@@ -0,0 +1,810 @@
+			state "humidity", label:'${currentValue}%'
+		}
+		valueTile("seen", "lastSeen", width: 2, height: 2, decoration: "flat") {
+			state "seen", label:'last seen ${currentValue}'
+		}
+
+		main("thermostatFull")
+		details([
+			"thermostatFull", 
+			"temperature","tempDown","tempUp",
+			"mode", "fanMode", "operatingState",
+			"heatingSetpoint", "heatDown", "heatUp",
+			"coolingSetpoint", "coolDown", "coolUp",
+			"toggle","program", "hold",
+			"humidity","refresh", "seen"
+
+		])
+	}
+	
+}
+
+def installed() {
+	log.debug "installed"
+	// checkInterval of 12 minutes.
+    sendEvent(name: "checkInterval", value: 60 * 12, displayed: false, data: [protocol: "zigbee", hubHardwareId: device.hub.hardwareID])
+
+	sendEvent(name: "temperature", value: 72, unit: "F")
+	sendEvent(name: "heatingSetpoint", value: 70, unit: "F")
+	sendEvent(name: "thermostatSetpoint", value: 70, unit: "F")
+	sendEvent(name: "coolingSetpoint", value: 76, unit: "F")
+	sendEvent(name: "thermostatMode", value: "off")
+	sendEvent(name: "thermostatFanMode", value: "fanAuto")
+	sendEvent(name: "thermostatOperatingState", value: "idle")
+	sendEvent(name: "humidity", value: 53, unit: "%")
+
+	state.refreshExtraAttributes = (state.refreshExtraAttributes?:0) - (25 * 60 * 60 * 1000)
+
+	runIn(2, "configure", [overwrite: true])  // Allow configure command to be sent and acknowledged before proceeding
+}
+
+def updated() {
+	log.debug "updated"
+    
+    // Make sure we poll all attributes from the device
+    state.refreshExtraAttributes = (state.refreshExtraAttributes?:0) - (25 * 60 * 60 * 1000)
+	// this may be the second call to configure, the overwrite should catch extra calls
+    runIn(2, "configure", [overwrite: true])
+}
+
+def refresh(){
+	log.trace "refreshing..."
+
+	def cmds = []
+		   cmds += zigbee.readAttribute(0x0405, 0x0000)
+		   cmds += zigbee.readAttribute(0x0000, 0x0005) 
+		   cmds += zigbee.readAttribute(0x0201, 0x0000) 
+		   cmds += zigbee.readAttribute(0x0201, 0x0011) 
+		   cmds += zigbee.readAttribute(0x0201, 0x0012) 
+		   cmds += zigbee.readAttribute(0x0201, 0x001C) 
+		   cmds += zigbee.readAttribute(0x0201, 0x001E) 
+		   cmds += zigbee.readAttribute(0x0204, 0x0001) 
+		   cmds += zigbee.readAttribute(0x0201, 0x0023) 
+		   cmds += zigbee.readAttribute(0x0201, 0x0024) 
+		   cmds += zigbee.readAttribute(0x0201, 0x0025) 
+		   cmds += zigbee.readAttribute(0x0201, 0x0029) 
+		   cmds += zigbee.readAttribute(0x0201, 0x0010, ["mfgCode": 0x1121]) 
+		   cmds += setThermostatTime()
+
+    def timeNow = new Date().time
+    if (!state.refreshExtraAttributes || (24 * 60 * 60 * 1000 < (timeNow - state.refreshExtraAttributes))) {
+        state.refreshExtraAttributes = timeNow
+
+		// also pull down the setpoint limits.
+        cmds += zigbee.readAttribute(0x0201, 0x0015)
+		cmds += zigbee.readAttribute(0x0201, 0x0016)
+		cmds += zigbee.readAttribute(0x0201, 0x0017)
+		cmds += zigbee.readAttribute(0x0201, 0x0018)
+    }
+	 
+	 sendZigbeeCmds(cmds, 100)
```

Comment:
We've have not seem any issues.  100ms should allow for 1.4KBytes of command text to the ZigBee chip.    Since our device radio is always-on shorter delays benefit UX.